docs/concepts/work-pools.md

```diff
@@ -320,6 +320,12 @@ Work queues can also have their own concurrency limits. Note that each queue is
 
 Together work queue priority and concurrency enable precise control over work. For example, a pool may have three queues: A "low" queue with priority `10` and no concurrency limit, a "high" queue with priority `5` and a concurrency limit of `3`, and a "critical" queue with priority `1` and a concurrency limit of `1`. This arrangement would enable a pattern in which there are two levels of priority, "high" and "low" for regularly scheduled flow runs, with the remaining "critical" queue for unplanned, urgent work, such as a backfill.
 
+Priority is evaluated if more is scheduled than can be submitted. This can be due to Concurrency limits that limit the number of flows running, or could be due to infrastructure limits such as Kubernetes where there is insufficient CPU and/or Memory availability. In these scenarios where work is in a `SCHEDULED` state, priority becomes relevant. 
```

Comment:
```suggestion
Priority is only evaluated if more runs are scheduled than can be submitted. This can be due to concurrency limits that limit the number of flows running or infrastructure limits such as insufficient CPU and/or memory availability on Kubernetes.
```

I don't think we actually do anything to handle CPU/memory availability though. We'll just create the jobs and Kubernetes will schedule them as needed.

Reply:
Also, priority still determines the _order_ in which runs are submitted even if they can all be submitted — right?

Reply:
For sure! I'll make those changes